src/AutoRest.CSharp.V3/Generation/Writers/RestClientWriter.cs

```diff
@@ -91,49 +91,55 @@ private void WriteClientCtor(CodeWriter writer, RestClient restClient, CSharpTyp
 
         private void WriteRequestCreation(CodeWriter writer, RestClientMethod operation)
         {
+            using var methodScope = writer.AmbientScope();
```

Comment:
We need an ambient scope around methods because parameter declarations appear before main method scope starts.

Reply:
Based on this, might make sense to go back to a code writer Method call, where it declares the parameters in the method's scope, instead of just ad-hoc appending them around parentheses like we are doing below. Not necessary for this PR, but something that would clean this and other methods up a bit.